substrate/frame/balances/src/lib.rs

```diff
@@ -287,14 +290,16 @@ pub mod pallet {
 			origin: OriginFor<T>,
 			dest: AccountIdLookupOf<T>,
 			#[pallet::compact] value: T::Balance,
+			memo: Option<T::Memo>,
 		) -> DispatchResultWithPostInfo {
```

Comment:
We should add a `transfer_with_memo` call at a high call index (like 50), to help keep the balance pallet compatible with most Substrate networks.